shared/chat/conversation/normal/index.native.tsx

```diff
@@ -35,39 +35,55 @@ const Conversation = React.memo((props: Props) => {
   const onLayout = React.useCallback((e: LayoutEvent) => {
     setMaxInputArea(e.nativeEvent.layout.height)
   }, [])
+  const innerComponent = (
+    <Kb.BoxGrow onLayout={onLayout}>
+      <Kb.Box2 direction="vertical" fullWidth={true} style={styles.innerContainer}>
+        <ThreadLoadStatus conversationIDKey={props.conversationIDKey} />
+        <PinnedMessage conversationIDKey={props.conversationIDKey} />
+        <ListArea
+          scrollListDownCounter={props.scrollListDownCounter}
+          scrollListToBottomCounter={props.scrollListToBottomCounter}
+          scrollListUpCounter={props.scrollListUpCounter}
+          onFocusInput={props.onFocusInput}
+          conversationIDKey={props.conversationIDKey}
+        />
+        {props.showLoader && <Kb.LoadingLine />}
+      </Kb.Box2>
+      <InvitationToBlock conversationID={props.conversationIDKey} />
+      <Banner conversationIDKey={props.conversationIDKey} />
+      <InputArea
+        focusInputCounter={props.focusInputCounter}
+        jumpToRecent={props.jumpToRecent}
+        onRequestScrollDown={props.onRequestScrollDown}
+        onRequestScrollToBottom={props.onRequestScrollToBottom}
+        onRequestScrollUp={props.onRequestScrollUp}
+        conversationIDKey={props.conversationIDKey}
+        maxInputArea={maxInputArea}
+      />
+    </Kb.BoxGrow>
+  )
   return (
-    <>
+    <Kb.Box style={styles.innerContainer}>
       <Kb.Box2 direction="vertical" fullWidth={true} fullHeight={true}>
         {props.threadLoadedOffline && <Offline />}
-        <HeaderArea conversationIDKey={props.conversationIDKey} />
-        <Kb.BoxGrow onLayout={onLayout}>
-          <Kb.Box2 direction="vertical" fullWidth={true} style={styles.innerContainer}>
-            <ThreadLoadStatus conversationIDKey={props.conversationIDKey} />
-            <PinnedMessage conversationIDKey={props.conversationIDKey} />
-            <ListArea
-              scrollListDownCounter={props.scrollListDownCounter}
-              scrollListToBottomCounter={props.scrollListToBottomCounter}
-              scrollListUpCounter={props.scrollListUpCounter}
-              onFocusInput={props.onFocusInput}
-              conversationIDKey={props.conversationIDKey}
-            />
-            {props.showLoader && <Kb.LoadingLine />}
-          </Kb.Box2>
-          <InvitationToBlock conversationID={props.conversationIDKey} />
-          <Banner conversationIDKey={props.conversationIDKey} />
-          <InputArea
-            focusInputCounter={props.focusInputCounter}
-            jumpToRecent={props.jumpToRecent}
-            onRequestScrollDown={props.onRequestScrollDown}
-            onRequestScrollToBottom={props.onRequestScrollToBottom}
-            onRequestScrollUp={props.onRequestScrollUp}
-            conversationIDKey={props.conversationIDKey}
-            maxInputArea={maxInputArea}
-          />
-        </Kb.BoxGrow>
+        {Styles.isTablet ? (
+          <Kb.KeyboardAvoidingView
```

Comment:
This extra KeyboardAvoidingView with a height offset is a tablet-only workaround for the screen not pushing itself high enough with an iPad keyboard.